typed-racket-lib/typed-racket/rep/base-types.rkt

```diff
@@ -0,0 +1,207 @@
+#lang racket/base
+
+(require "../utils/utils.rkt"
+         (utils hset)
+         (rep rep-utils base-type-rep type-mask core-rep)
+         racket/undefined
+         (types numeric-predicates)
+         racket/extflonum
+         ;; for base type contracts and predicates
+         ;; use '#%place to avoid the other dependencies of `racket/place`
+         (for-template
+           racket/base
+           racket/contract/base
+           racket/undefined
+           racket/extflonum
+           (only-in racket/pretty pretty-print-style-table?)
```

Comment:
We should consider a submodule for that definition so that we need fewer deps here.